scripts/linters/general_purpose_linter.py

```diff
@@ -748,6 +747,43 @@ def _check_bad_patterns(self):
                 python_utils.PRINT(summary_message)
         return summary_messages
 
+    def _check_newline_at_eof(self):
+        """This function is used to detect newline at the end of file."""
+        if self.verbose_mode_enabled:
+            python_utils.PRINT(
+                'Starting newline at eof check\n'
+                '----------------------------------------')
+        summary_messages = []
+        files_to_lint = self.all_filepaths
+        failed = False
+
+        with linter_utils.redirect_stdout(sys.stdout):
+            for filepath in files_to_lint:
+                file_content = FILE_CACHE.readlines(filepath)
+                file_length = len(file_content)
+                if (
+                        file_length >= 2 and
```

Comment:
Why this checked is needed `file_length >=2`?

Reply:
This is needed to check if the file is not empty. Changed it to `file_length >=1`